### ISSUE TITLE
configurable n-window

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc-ui#864]
Ideally the user would be able to set the `n-window` for each view independently.

A good use case would be an operator working on a complex workflow. They might want the tree view in `n=1` and the graph view in `n=0`.

One day we would be able to construct an n-window around an arbitrary selection of start tasks which would enhance this further.

E.G. The operator could have the tree view open to `n=1` and the graph view to an `n=2` based around the tasks they are selecting in the tree view.

This would be useful in understanding the impacts of task failures, delays, bottlenecks, etc.

This is a complex thing to do and would require server-side assistance. I think the data store currently supports the concept of an arbitrary n-window selection.

Related:

* https://github.com/cylc/cylc-ui/issues/1093
* https://github.com/cylc/cylc-ui/issues/434
* https://github.com/cylc/cylc-ui/issues/534

**Pull requests welcome!**

**Comment by hjoliver** [comment]
> One day we would be able to construct an n-window around an arbitrary selection of start tasks which would enhance this further.

Yeah that's definitely on the wanted list.

**Comment by oliver-sanders** [comment]
> One day we would be able to construct an n-window around an arbitrary selection of start tasks which would enhance this further.

Note: This will require data provision on the backend (the same provision required for `n>1` windows).

**Comment by oliver-sanders** [comment]
The n-window also has a role in managing "offline" (historical) task/job information since `n=cycle` e.g. `n=20200101` means "all tasks from the cycle 20200101 irrespective of their edge distance from active tasks".

See also: https://github.com/cylc/cylc-uiserver/issues/378

**Comment by oliver-sanders** [comment]
Blocked by https://github.com/cylc/cylc-flow/issues/5609

**Comment by ColemanTom** [comment]
Just looking through and saw this one. The Issue you had blocking this is closed, is this still blocked?

**Comment by hjoliver** [comment]
Thanks for the heads-up, this should be unblocked now, but @oliver-sanders can confirm.

**Comment by oliver-sanders** [comment]
Yes this is no longer blocked. It should be relatively easy to get working, I got a prototype working in a few lines.

**Comment by steoxley** [comment]
This would be really useful in a rose-stem context which doesn't cycle and that people want to repeatedly rerun certain tasks that may otherwise would have fallen out of scope.

**Comment by oliver-sanders** [comment]
This will be very useful for most contexts, unfortunately, we haven't had the time to work on it so far.
